.github/workflows/docs.yml

```diff
@@ -0,0 +1,187 @@
+      - name: Generate TeX document files
+        run: |
+          make clean
+          make docs
+          mkdir -p $DOCS_DIR
+          mv relay-states.pdf $DOCS_DIR
+
+      # A push event is a master merge; deploy to primary bucket.
+      - if: github.event_name == 'push'
+        name: Upload generated files
+        uses: thesis/gcp-storage-bucket-action@v3.1.0
+        with:
+          service-key: ${{ secrets.DOCS_KEEP_NETWORK_UPLOADER_SERVICE_KEY_JSON }}
+          project: keep-test
+          bucket-name: docs.keep.network
+          bucket-path: core-wip2 # TODO: remove `-wip2` after tests
+          build-folder: tmp/docs
+
+      # A pull_request event is a PR; deploy to preview bucket.
+      - if: github.event_name == 'pull_request'
+        name: Upload preview of generated files
+        uses: thesis/gcp-storage-bucket-action@v3.1.0
+        with:
+          service-key: ${{ secrets.DOCS_KEEP_NETWORK_UPLOADER_SERVICE_KEY_JSON }}
+          project: keep-test
+          bucket-name: docs.keep.network
+          bucket-path: core/${{ github.head_ref }}-wip2 # TODO: remove `-wip2` after tests
+          build-folder: tmp/docs
+    
+  docs-solidity:
+    runs-on: ubuntu-latest
+    needs: docs-detect-changes
+    if: |
+      github.event_name != 'pull_request'
+        || needs.docs-detect-changes.outputs.path-filter == 'true'
+    env:
+      SOLIDITY_DOCS_DIR: ${{ github.workspace }}/docs/solidity
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Generate solidity docs
+        working-directory: ./solidity
+        run: |
+          npm ci
+          npx truffle compile
+          mkdir -p $SOLIDITY_DOCS_DIR
+          node scripts/generate-api-docs.js > $SOLIDITY_DOCS_DIR/index.adoc
+
+      - name: Upload Solidity Docs Artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: Solidity Docs
+          path: ${{ env.SOLIDITY_DOCS_DIR }}/index.adoc
+
+  docs-html:
+    if: always()
+    runs-on: ubuntu-latest
+    needs: [docs-detect-changes, docs-solidity]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Download Solidity Docs Artifact
+        uses: actions/download-artifact@v2
+        continue-on-error: true
+        with:
+          name: Solidity Docs
+          path: ${{ github.workspace }}/docs/solidity
+
+      - name: Build HTML docs
+        id: html
+        uses: thesis/asciidoctor-action@v1.0.1
+        with:
+          files: 'docs/*.adoc docs/**/*.adoc'
+          args: '-a revdate=`date +%Y-%m-%d` --failure-level=ERROR'
+
+      # A push event is a master merge; deploy to primary bucket.
+      - if: github.event_name == 'push'
+        name: Upload asciidocs
+        uses: thesis/gcp-storage-bucket-action@v3.1.0
+        with:
+          service-key: ${{ secrets.DOCS_KEEP_NETWORK_UPLOADER_SERVICE_KEY_JSON }}
+          project: keep-test
+          bucket-name: docs.keep.network
+          bucket-path: core-wip2 # TODO: remove `-wip2` after tests
```

Comment:
I was under an impression that I already commented on it but looks like the comment was not submitted, so... 
For keep-core docs, we're probably going to push it to the root directory of the bucket. I believe we do that now. But would love to hear @Shadowfiend thoughts around that.

Reply:
Yes, in one of the previous comments I also was considering dropping `core` in `bucket-path`, as in CircleCI job we are currently uploading files directly to `cfc-production` project's `docs.keep.network` bucket.